plugin/pkg/admission/priority/admission.go

```diff
@@ -157,11 +161,21 @@ func (p *priorityPlugin) admitPod(a admission.Attributes) error {
 		return errors.NewBadRequest("resource was marked with kind Pod but was unable to be converted")
 	}
 
-	// Make sure that the client has not set `priority` at the time of pod creation.
-	if operation == admission.Create && pod.Spec.Priority != nil {
-		return admission.NewForbidden(a, fmt.Errorf("the integer value of priority must not be provided in pod spec. Priority admission controller populates the value from the given PriorityClass name"))
+	if operation == admission.Update {
+		oldPod, ok := a.GetOldObject().(*api.Pod)
+		if !ok {
+			return errors.NewBadRequest("resource was marked with kind Pod but was unable to be converted")
+		}
+
+		// This admission plugin set pod.Spec.Priority on create.
+		// Ensure the existing priority is preserved on update.
+		if pod.Spec.Priority == nil && oldPod.Spec.Priority != nil {
+			pod.Spec.Priority = oldPod.Spec.Priority
+		}
```

Comment:
If `pod.Spec.Priority != oldPod.Spec.Priority` shouldn't we throw an error. It means user tried to update priority which is wrong. This field is immutable.

Reply:
API validation already disallows modification. this is preserving the existing value if the incoming pod dropped the field entirely (new pod has it set to nil).

that would occur if a client was using a back-level go struct to get/update, or did `kubectl create -f pod.yaml && kubectl replace -f pod.yaml`

Reply:
the existing test failures on #65722 demonstrate how a client unaware of priority can be bitten by enabling an admission plugin which auto-sets it on create, and doesn't do correspondingly helpful things on update

Reply:
Thanks @liggitt. One more case - What would happen, if we have 2 priorityClasses with same values. We are setting priority value only here, shouldn't we set priorityClass as well.

Reply:
> What would happen, if we have 2 priorityClasses with same values.

the admission plugin simply ensures the priority value it resolved to is preserved.

if the user attempts to change the priority class name on update, API validation will prevent that as a disallowed change.